### PR TITLE
Support for reading & writing HEIF files format using pillow_heif

### LINKED ISSUE #1503: Accepting HEIC/HEIF as image input and output
### Thumbor request URL
[HEIC](http://localhost:8888/unsafe/https://github.com/alexcorvi/heic2any/blob/master/demo/1.heic?raw=true)

### Expected behaviour
Thumbor should accept `HEIC` and `HEIF` as image input and output.
Why should this format be used?
```
HEIC/HEIF not only reduces file size by ~50% compared to JPEG, but with iOS 11, it now is also 
the default image format for more than 50% of cell phones in the west
```

### Actual behaviour
Nowadays, Thumbor doesn't accept `HEIC` and `HEIF` as image input or output.

### PR DESCRIPTION
**Pull request by bigcat88**
Resolves #1503

### CODE REVIEW

File: thumbor/engines/__init__.py [hunk 1]
```diff
@@ -129,6 +129,8 @@ def get_mimetype(cls, buffer):
             img_mime = "image/jp2"
         elif buffer[4:12] in (b"ftypavif", b"ftypavis"):
             img_mime = "image/avif"
+        elif buffer[4:12] in (b"ftypheic", b"ftypmif1"):
```
Comment (RaphaelVRossi) [inline thumbor/engines/__init__.py:132]: Hey @bigcat88, Why you didn't validate other magic number for `HEIC/HEIF`? Can you explain to me, please?!

Like [here](https://github.com/bigcat88/pillow_heif/blob/3ffc497617da6c45a1bbaf7cf59e4107ac047df6/pillow_heif/as_opener.py#L120) 

```python
if magic[8:12] in (
        b"heic",
        b"heix",
        b"heim",
        b"heis",
        b"hevc",
        b"hevx",
        b"hevm",
        b"hevs",
        b"mif1",
        b"msf1",
    ):
```

**Comment by RaphaelVRossi** [comment]
Thanks for your contribution to Thumbor 🥳

File: thumbor/engines/__init__.py (hunk 1 above)
Comment (bigcat88) [reply thumbor/engines/__init__.py:132]: Maybe `heix` can be added, it is a `10 bit` Canon cameras format with `.hif` extension.
I never seen other identifiers and not sure that libheif will be able to parse them correctly.
`hevc`, `hevx`, `hevm`, `hevs` is a video formats - for now it is better to skip them.

Comment (RaphaelVRossi) [reply thumbor/engines/__init__.py]: Nice! Thanks for the explanation, and the updates!

**Comment by RaphaelVRossi** [comment]
@heynemann @scorphus Hey Guys! Could you review this PR, please?

**Comment by RaphaelVRossi** [comment]
Hey @bigcat88, I was testing your changes, and found a little bug when Thumbor doesn't find `pillow-heif/pillow-avif-plugin` installed, and try to change format to `.jpg`, and this breaks when the original image is PNG / WEBP.

I made a little fix, maybe you can change the same in your PR.

See [this](https://github.com/thumbor/thumbor/pull/1512/files#diff-cbfef243ebe9c91836cec802a2a83bc7dfb089be87d71ddcee5901ae1ad5022fR243)

**Comment by RaphaelVRossi** [comment]
LGTM!

**Pull request merged**